pkg/kapp/resourcesmisc/apps_v1_deployment_test.go

```diff
@@ -59,6 +59,58 @@ status:
 	require.Equal(t, expectedState, state, "Found incorrect state")
 }
 
+func TestAppsV1DeploymentReplicaFailure(t *testing.T) {
+
+	depYAML := `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
```

Comment:
You also need to add metadata.generation (which should be the same as status.observedGeneration).
You can use `./hack/test.sh` to run the unit tests locally.